src/java.base/share/classes/java/util/random/RandomGeneratorFactory.java

```diff
@@ -0,0 +1,524 @@
+ * or visit www.oracle.com if you need additional information or have any
+ * questions.
+ */
+
+package java.util.random;
+
+import java.lang.reflect.Constructor;
+import java.lang.reflect.Method;
+import java.math.BigInteger;
+import java.security.AccessController;
+import java.security.PrivilegedActionException;
+import java.security.PrivilegedExceptionAction;
+import java.util.function.Function;
+import java.util.Map;
+import java.util.random.RandomGenerator.ArbitrarilyJumpableGenerator;
+import java.util.random.RandomGenerator.JumpableGenerator;
+import java.util.random.RandomGenerator.LeapableGenerator;
+import java.util.random.RandomGenerator.SplittableGenerator;
+import java.util.random.RandomGenerator.StreamableGenerator;
+import java.util.ServiceLoader;
+import java.util.ServiceLoader.Provider;
+import java.util.stream.Collectors;
+import java.util.stream.Stream;
+import jdk.internal.util.random.RandomSupport.RandomGeneratorProperty;
+
+/**
+ * This is a factory class for generating multiple random number generators
+ * of a specific algorithm. {@link RandomGeneratorFactory} also provides
+ * methods for selecting random number generator algorithms.
+ *
+ * A specific {@link RandomGeneratorFactory} can be located by using the
+ * {@link RandomGenerator#factoryOf(String)} method, where the argument string
+ * is the name of the algorithm required. The method
+ * {@link RandomGenerator#all()} produces a {@link Stream} of all available
+ * {@link RandomGenerator RandomGenerators} that can be searched to locate a
+ * {@link RandomGeneratorFactory} suitable to the task.
+ *
+ * There are three methods for constructing a RandomGenerator instance,
+ * depending on the type of initial seed required.
+ * {@link RandomGeneratorFactory#create(long)} is used for long
+ * seed construction,
+ * {@link RandomGeneratorFactory#create(byte[])} is used for byte[]
+ * seed construction, and
+ * {@link RandomGeneratorFactory#create()} is used for random seed
+ * construction. Example;
+ *
+ * <pre>{@code
+ *    RandomGeneratorFactory<RandomGenerator> factory = RandomGenerator.factoryOf("Random");
+ *
+ *     for (int i = 0; i < 10; i++) {
+ *         new Thread(() -> {
+ *             RandomGenerator random = factory.create(100L);
+ *             System.out.println(random.nextDouble());
+ *         }).start();
+ *     }
+ * }</pre>
+ *
+ * RandomGeneratorFactory also provides methods describing the attributes (or properties)
+ * of a generator and can be used to select random number generator algorithms.
+ * These methods are typically used in
+ * conjunction with {@link RandomGenerator#all()}. In this example, the code locates the
+ * {@link RandomGeneratorFactory} that produces {@link RandomGenerator RandomGenerators}
+ * with the highest number of state bits.
+ *
+ * <pre>{@code
+ *     RandomGeneratorFactory<RandomGenerator> best = RandomGenerator.all()
+ *         .sorted((f, g) -> Integer.compare(g.stateBits(), f.stateBits()))
```

Comment:
use Comparator.comparingInt(RandomGenerator::stateBits) instead of the lambda

Reply:
Not sure that `.sorted(Comparator.comparingInt(RandomGeneratorFactory<RandomGenerator>::stateBits).reversed())` is simpler than  `.sorted((f, g) -> Integer.compare(g.stateBits(), f.stateBits()))`.

Reply:
At least, it's more clear that it's reversed, i've initially miss the fact that f and g are swapped.
And :: is able to do inference so, i believe it can be written
  `.sorted(Comparator.comparingInt(RandomGeneratorFactory::stateBits).reversed())`

Reply:
Unfortunately it couldn't be inferred

Reply:
It's because you have added reverse() as a postfix operation so the inference can not flow backward as it should,
using Collections.reverseOrder() should fix that
`.sorted(Collections.reverseOrder(Comparator.comparingInt(RandomGeneratorFactory::stateBits)))`

using some import statics for reverseOrder and comparintInt make the code readable but given it's in the doc, we are out of luck here.